src/www/app/outline_server.ts

```diff
@@ -31,8 +31,9 @@ export class OutlineServer implements Server {
   errorMessageId?: string;
 
   constructor(
-      public readonly id: string, private readonly config: ShadowsocksConfig,
-      private tunnel: Tunnel, private eventQueue: events.EventQueue) {
+      public readonly id: string, public readonly accessKey: string,
```

Comment:
It's confusing to have both the accessKey and the config.
Can't we have the same paramters as OutlineServerJson?
Pass id, name, and  accessKey (besides tunnel and eventQueue).
We need to change the server factory accordingly too.

I think we can ditch ShadowsocksConfig for the most part, and have something specific for the proxying.

We have a design problem with the factory.  We should eventually get rid of that, this code should  control that. Instead we could be injecting a tunnel factory instead, but that a bit of a bigger change we can do later.

Reply:
Done in 3e18673. I updated the server factory and `OutlineServer` constructor per your suggestion. I think it's clearer now, at the expense of parsing the access key twice, since the server repository needs to flag unsupported ciphers for backwards compatibility.

I kept the serve name getter/setter in order to update `config.name`, which is used by native code.

> We have a design problem with the factory. 

I have upcoming changes that will refactor the server factory into an interface that will allow native platforms to create only the tunnel.